grpcclient/lc_client.go

```diff
@@ -33,15 +34,15 @@ import (
 	"context"
 
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
```

Comment:
Does someone see value in adding later a custom Circuit Breaker middleware? I had a look into the grpc-ecosystem repo and didn't seem to find one.

Reply:
afaik `circuit breaker` is a service, which holds certain state, so it cant be a middleware

Reply:
It does indeed keep a value with 3 possible states. CLOSED, OPEN, HALF-OPEN. This can be safely kept in memory and is not considered as making the code stateful. It does not lead to inconsistencies, the initial state is CLOSED.

Reply:
this is a library which will be compiled into a client. What is a benefit for each of thousands of clients making concurrent request to have their own settings? This setting wont be shared across clients

Reply:
> The state doesn't need to be shared across thousands of clients, every instance (thread, process, machine) manages its own state
I think you don't really understand what `Circuit breaker` is. The article states
```
In a multi-node (clustered) server, the state of the upstream service will need to be reflected across all the nodes in the cluster. Therefore, implementations may need to use a persistent storage layer, e.g. a network cache such as [Memcached](https://en.wikipedia.org/wiki/Memcached) or [Redis](https://en.wikipedia.org/wiki/Redis), or local cache (disk or memory based) to record the availability of what is, to the application, an external service.
```
It is meant to be implemented on the server side, whereas current PR and current package will be used on the client, it's middleware doesn't need to  implement `Circuit breaker`